### PR TITLE
Adding test coverage

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

Adding Coverage reports

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by gpoleze — APPROVED** [review]
👍

File: .github/workflows/codeclimate_api.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    env:
+      CC_TEST_REPORTER_ID: 49f511c02b4a733d0209c387cda806284462e2cdbb5393ea75426f9e55c21169
+
+    name: Create, format and upload Java coverage artifact
+    runs-on: ubuntu-latest
+
+    steps:
+
+      # Checkout Project
+      - name: Checkout Courtlist Repository
```
Comment (BrendanBeachBC) [inline .github/workflows/codeclimate_api.yml:20]: Copy paste error, Courtlist

File: .github/workflows/codeclimate_api.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        run: |
+          echo "FILE_COUNTER=$(ls -1q ./coverage | wc -l )" >> $GITHUB_ENV
+          #echo "::set-env name=FILE_COUNTER::$(ls -1q ./coverage | wc -l )"
+    
+      # Upload JSON for debugging purposes
+      - name: Upload JSON for debugging purposes
+        uses: actions/upload-artifact@v2
+        with:
+          name: summed-java-coverage-report
+          path: coverage/courtlist-codeclimate.json
```
Comment (BrendanBeachBC) [inline .github/workflows/codeclimate_api.yml:89]: Same courtlist name as above

File: .github/workflows/codeclimate_api.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    
+      # Upload total coverage report to Code Climate
+      - name: Upload coverage report to Code Climate
+        run: ./cc-test-reporter upload-coverage -i coverage/courtlist-codeclimate.json
```
Comment (BrendanBeachBC) [inline .github/workflows/codeclimate_api.yml:93]: Same courtlist name as above

**Review by BrendanBeachBC — CHANGES_REQUESTED** [review]
Looks good, but a couple of copy paste naming issues I noticed

File: .github/workflows/codeclimate_api.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+     
+      # Run Maven Verify to generate all jacoco reports
+      - name: Build with Maven
+        run: mvn -B verify --file src/ecrc-api/pom.xml
+
+
+      # Formatting the coverage reports generated (dynamically)
+      - name: Format coverage reports
+        run: |
+          JACOCO_SOURCE_PATH=/src/ecrc-api/ ./cc-test-reporter format-coverage target/site/jacoco/jacoco.xml --input-type jacoco --output coverage/courtlist-codeclimate.json;
```
Comment (BrendanBeachBC) [inline .github/workflows/codeclimate_api.yml:76]: Reference to courtlist in the json name

**Pull request merged**